src/plugins/vis_type_vislib/public/vislib/lib/axis/axis_scale.js

```diff
@@ -49,10 +49,20 @@ export class AxisScale {
     });
   }
 
-  getTimeDomain(data) {
+  getDomainExtent(data) {
     return [this.minExtent(data), this.maxExtent(data)];
   }
 
+  getOrdinalDomain(values = []) {
+    if (this.ordered?.interval !== undefined) {
+      const [min, max] = this.getDomainExtent(values);
+      const valuesInterval = Math.ceil((max - min) / values.length);
+      return range(min, max + valuesInterval, valuesInterval);
+    }
+
+    return values;
+  }
```

Comment:
This doesn't seem to change the current behviour. Could you actually remove this logic in place of just `this.values` (as it is now)? I'll create a PR for this after this is merged.

Talking with macro this logic is not the intended effect. See https://github.com/elastic/kibana/issues/88693#issuecomment-763004009